extensions/cornerstone/src/services/ViewportService/CornerstoneViewportService.ts

```diff
@@ -874,28 +879,35 @@ class CornerstoneViewportService extends PubSubService implements IViewportServi
     });
 
     // For SEG and RT viewports
-    const { addOverlayFn, imageIds } = this._processExtraDisplaySetsForViewport(viewport) || {};
-
-    if (!filteredVolumeInputArray.length && addOverlayFn) {
-      // if there is no volume input array, and there is an addOverlayFn, means we need to take
-      // care of the background overlay display set first then the addOverlayFn will add the
-      // SEG displaySet
-      const sampleImageId = imageIds[0];
-      const backgroundDisplaySet = displaySetService.getDisplaySetsBy(
-        displaySet =>
-          !displaySet.isOverlayDisplaySet &&
-          displaySet.images.some(image => image.imageId === sampleImageId)
-      );
-
-      if (backgroundDisplaySet.length !== 1) {
-        throw new Error('Background display set not found');
-      }
+    const overlayProcessingResults = this._processExtraDisplaySetsForViewport(viewport) || [];
+    if (!filteredVolumeInputArray.length && overlayProcessingResults?.length) {
```

Comment:
Does this make sense for volume viewports?

Reply:
sure